src/shared/logger.ts

```diff
@@ -0,0 +1,82 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as vscode from 'vscode'
+import * as winston from 'winston'
+
+let logSettings: LogSettings
+
+class LogSettings {
+    private readonly outputChannel: vscode.OutputChannel
+    private readonly logger: winston.Logger
+    private readonly logPath: string
+
+    public constructor(outputChannel: vscode.OutputChannel, logPath: string, level: string = 'info') {
+        this.outputChannel = outputChannel
+        this.logger = winston.createLogger({
+            transports: [new winston.transports.File({ filename: logPath })]
+        })
+        this.logPath = logPath
+        // TODO: check if a log level exists--we can add a field for this to the settings page.
+        this.setLevel(level)
+    }
+
+    public getOutputChannel(): vscode.OutputChannel {
+        return this.outputChannel
+    }
+
+    public getLogger(): winston.Logger {
+        return this.logger
+    }
+
+    public setLevel(level: string): void {
+        // TODO: Ensure log level is valid
```

Comment:
If you change the type from `string` to `'info' | 'warn' | 'error'`, the compiler will validate this for us.

Reply:
Looks like this is on an outdated version, current version validates against `winston`'s set levels: https://github.com/aws/aws-toolkit-vscode/pull/302/files#diff-4fdc113ff641e6c27c6f66c0d90c90b8R40